curl/src/main/scala/org/http4s/curl/unsafe/libcurl.scala

```diff
@@ -80,9 +81,15 @@ private[curl] object libcurl_const {
 private[curl] object libcurl {
 
   type CURL
-  type CURLcode = CInt
+  final case class CURLcode(value: CInt) extends AnyVal {
```

Comment:
Oops! This broke the scala 3 build, doesn't scala native support value classes in scala 3?
Or I'm doing something wrong?

Reply:
You cannot nest it inside of the `libcurl` extern object, I think.

Reply:
Moving out class definition solved the problem. :sweat_smile: